src/hotspot/share/runtime/thread.cpp

```diff
@@ -3417,7 +3417,7 @@ void Threads::destroy_vm() {
     // queue until after the vm thread is dead. After this point,
     // we'll never emerge out of the safepoint before the VM exits.
 
-    MutexLocker ml(Heap_lock, Mutex::_no_safepoint_check_flag);
+    MutexLocker ml(Heap_lock);
```

Comment:
Can you add an assert before this that the thread is_terminated() as that is the key to the mutex acquisition not doing a safepoint check.

Reply:
Ok, I can add:
+    assert(thread->is_terminated(), "must be terminated before acquiring Heap_lock");
and rerun tests.